# Estimation And Confidence Intervals

In [1]:
# importing the libraries
import numpy as np 
import pandas as pd
import scipy.stats as st
# sample data
x=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
df=pd.DataFrame(x) # convert the given list of sample to Data Frame
df.columns=['sample data']
df

sample data
0          1.13
1          1.55
2          1.43
3          0.92
4          1.25
5          1.36
6          1.32
7          0.85
8          1.07
9          1.48
10         1.20
11         1.33
12         1.18
13         1.22
14         1.29

In [2]:
df.describe() # To describe the data frame

sample data
count    15.000000
mean      1.238667
std       0.193164
min       0.850000
25%       1.155000
50%       1.250000
75%       1.345000
max       1.550000

In [3]:
x_bar=df.mean() # mean of the sample data 
x_bar

sample data    1.238667
dtype: float64

In [4]:
s=df.std() # sample standard deviation
s

sample data    0.193164
dtype: float64

In [5]:
n=df.count() # length of the sample
n

sample data    15
dtype: int64

In [10]:
# t-Distribution for sample standard deviation
st.t.interval(0.99,n-1,loc=x_bar,scale=s/np.sqrt(n)) #scale is standard error

(array([1.09019734]), array([1.38713599]))

while using the sample standard deviation ,We are 99% confident that the true average durability of print-heads lies between 1.09019734 million and 1.38713599 million characters before failure.

In [19]:
t=st.t.ppf(1-(0.01)/2,n-1)
t

array([2.97684273])

In [26]:
me=t*s/np.sqrt(n) # Margin of error which is error from the sample mean 
me

sample data    0.148469
dtype: float64

In [29]:
x_bar-me,x_bar+me #manual way 

(sample data    1.090197
 dtype: float64,
 sample data    1.387136
 dtype: float64)

In [25]:
# z-Distribution
sigma=0.2 # population satandard deviation 
st.norm.interval(0.99,loc=x_bar,scale=sigma/np.sqrt(n)) # z distribution

(array([1.10565141]), array([1.37168192]))

while using population standard deviation ,We are 99% confident that the true average durability of print-heads lies between 1.10565141 million and 1.37168192 million characters before failure .

In [22]:
st.norm.ppf(1-(0.01/2))  # To  get the values of in z table

np.float64(2.5758293035489004)